helm/nessie/templates/httproute.yaml

```diff
@@ -0,0 +1,114 @@
+{{- $defaultBackendPort := ((include "nessie.service.primaryPort" .) | trim) | int -}}
+{{- $routes := .Values.gatewayApi.httpRoutes | default (list) -}}
+{{- range $idx, $route := $routes }}
+{{- $routeName := default (printf "%s%s" (include "nessie.fullname" $) (default "" $route.nameSuffix)) $route.name -}}
+{{- $routeNamespace := default $.Release.Namespace $route.namespace -}}
+{{- $rules := $route.rules | default (list) -}}
+{{- if not (gt (len $rules) 0) -}}
+{{- fail "gatewayApi.httpRoutes[].rules must contain at least one rule" -}}
+{{- end -}}
+{{- if gt $idx 0 }}
+---
+{{- end }}
+apiVersion: {{ $httpRouteApiVersion }}
+kind: HTTPRoute
+metadata:
+  name: {{ $routeName }}
+  namespace: {{ $routeNamespace }}
+  labels:
+    {{- include "nessie.labels" $ | nindent 4 }}
+    {{- with $route.labels }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
+  {{- with $route.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  {{ with $route.hostnames }}
+  hostnames:
+    {{- range . }}
+    - {{ . | quote }}
+    {{- end }}
+  {{- end }}
+  {{- if $route.parentRefs }}
+  parentRefs:
+    {{ toYaml $route.parentRefs | nindent 4 }}
+  {{- else }}
+  parentRefs:
+    - name: {{ $gatewayName }}
+      namespace: {{ $gatewayNamespace }}
+  {{- end }}
+  rules:
+{{- range $rule := $rules }}
+    - {{- if $rule.matches }}
+      matches:
+        {{- toYaml $rule.matches | nindent 8 }}
+      {{- end }}
+      {{ with $rule.filters }}
+      filters:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- $backends := $rule.backendRefs | default (list) }}
+      {{- if not (gt (len $backends) 0) -}}
+      {{- fail "gatewayApi.httpRoutes[].rules[].backendRefs must contain at least one backend" -}}
+      {{- end }}
+      backendRefs:
+{{- range $backend := $backends }}
+        {{- $service := $backend.service | default (dict) -}}
+        {{- $computedServiceName := default (printf "%s%s" (include "nessie.fullname" $) (default "" $service.nameSuffix)) $service.name -}}
+        {{- $explicitName := default "" $backend.name -}}
+        {{- $name := default $computedServiceName $explicitName -}}
+        {{- $name = required "gatewayApi.httpRoutes[].rules[].backendRefs[] requires a backend name" $name -}}
+        {{- $serviceNamespace := default $routeNamespace $service.namespace -}}
+        {{- $backendNamespace := default $serviceNamespace $backend.namespace -}}
+        {{- $port := default (default $service.port $defaultBackendPort) $backend.port -}}
+        {{- $kind := "" -}}
+        {{- if hasKey $backend "kind" -}}
+        {{- $kind = $backend.kind -}}
+        {{- else -}}
+        {{- $kind = "Service" -}}
+        {{- end }}
+        - name: {{ $name }}
+          {{- if $backendNamespace }}
+          namespace: {{ $backendNamespace }}
+          {{- end }}
+          {{- with $backend.group }}
+          group: {{ . }}
+          {{- end }}
+          {{- if $kind }}
```

Comment:
It seems `$kind` is always defined and non-empty.

Reply:
Fixed